src/FloatImageViewer.hpp

```diff
@@ -121,12 +154,18 @@ class FloatImageViewer : public QQuickItem
     QSharedPointer<FloatImage> _image;
     QRectF _boundingRect;
     QSize _textureSize;
-    QSize _sourceSize = QSize(0,0);
+    QSize _sourceSize = QSize(0, 0);
 
     QVariantMap _metadata;
+
+    Surface _surface;
+    // Prevent to update surface without the root created
+    bool _createRoot = true;
+    // Level of downscale for images of a Panorama
+    int _downscaleLevel = 4;
```

Comment:
```suggestion
    int _downscaleLevel = 0;
```

And should be set to a bigger value only in the Panorama.